_examples/json/main.go

```diff
@@ -0,0 +1,86 @@
+// nolint: golint, dupl
+package main
+
+import (
+	"os"
+
+	"github.com/alecthomas/kong"
+
+	"github.com/alecthomas/participle/v2"
+	"github.com/alecthomas/participle/v2/lexer"
+)
+
+var (
+	jsonLexer = lexer.MustSimple([]lexer.SimpleRule{
+		{Name: "Comment", Pattern: `\/\/[^\n]*`},
+		{Name: "String", Pattern: `"(\\"|[^"])*"`},
+		{Name: "Number", Pattern: `[-+]?(\d*\.)?\d+`},
+		{Name: "Punct", Pattern: `[-[!@#$%^&*()+_={}\|:;"'<,>.?/]|]`},
+		{Name: "Null", Pattern: "null"},
+		{Name: "True", Pattern: "true"},
+		{Name: "False", Pattern: "false"},
+		{Name: "EOL", Pattern: `[\n\r]+`},
+		{Name: "Whitespace", Pattern: `[ \t\n\r]+`},
```

Comment:
`\n\r` will never match here because `EOL` takes precedence.

Reply:
fixed it